components/director/internal/domain/bundleinstanceauth/resolver.go

```diff
@@ -268,3 +272,96 @@ func (r *Resolver) RequestBundleInstanceAuthDeletion(ctx context.Context, authID
 
 	return r.conv.ToGraphQL(instanceAuth)
 }
+
+// CreateBundleInstanceAuth creates a BundleInstanceAuth for a Bundle with ID - bundleID from a given BundleInstanceAuthCreateInput
+func (r *Resolver) CreateBundleInstanceAuth(ctx context.Context, bundleID string, in graphql.BundleInstanceAuthCreateInput) (*graphql.BundleInstanceAuth, error) {
+	tx, err := r.transact.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	defer r.transact.RollbackUnlessCommitted(ctx, tx)
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	log.C(ctx).Infof("Creating BundleInstanceAuth for Bundle with id %q", bundleID)
+
+	bndl, err := r.bndlSvc.Get(ctx, bundleID)
+	if err != nil {
+		return nil, err
+	}
+
+	convertedIn, err := r.conv.CreateInputFromGraphQL(in)
+	if err != nil {
+		return nil, err
+	}
+
+	instanceAuthID, err := r.svc.CreateBundleInstanceAuth(ctx, bundleID, convertedIn, bndl.InstanceAuthRequestInputSchema)
+	if err != nil {
+		return nil, err
+	}
+
+	instanceAuth, err := r.svc.Get(ctx, instanceAuthID)
+	if err != nil {
+		return nil, err
+	}
+	log.C(ctx).Infof("Successfully created BundleInstanceAuth with id %q for Bundle with id %q", instanceAuthID, bundleID)
```

Comment:
Can we move the log after the transaction commiting as it will be misleading in case the trainsaction fails to commit